### PR TITLE
Implement darkmode on books,exercises and packages search result

### PR DESCRIPTION
**Pull request by oyenuga17**
@SaySayo please help review

### CODE REVIEW

File: asset/css/doc.css [hunk 1]
```diff
@@ -203,9 +203,6 @@ div.odoc .comment-delim {
 }
 
 .navmap-tag {
-  display:flex;
-  align-items: center;
-  justify-content: center;
```
Comment (sabine) [inline asset/css/doc.css]: is this an intentional change? looks unrelated to dark mode and should probably not be in this patch

File: src/ocamlorg_frontend/components/navmap.eml [hunk 2]
```diff
@@ -28,7 +28,7 @@ let kind_title = function
   | Class_type -> "Class type"
   | _ -> "?"
 
-let title_style = "flex-1 flex-nowrap py-1 md:py-0.5 pr-1 text-legacy-lighter"
+let title_style = "flex-1 flex-nowrap py-1 md:py-0.5 pr-1  text-title dark:text-dark-title"
```
Comment (sabine) [inline src/ocamlorg_frontend/components/navmap.eml]:
```suggestion
let title_style = "flex-1 flex-nowrap py-1 md:py-0.5 pr-1 text-title dark:text-dark-title"
```

File: src/ocamlorg_frontend/pages/exercises.eml [hunk 3]
```diff
@@ -154,4 +160,47 @@ Learn_layout.three_column_layout
...
+            scrollY: window.scrollY,
+            sectionYPositions: null,
+  
+            isWithin,
+            computeSectionYPositions,
+          }
+        ))
+      })
+    </script>
+  </div>
```
Comment (sabine) [inline src/ocamlorg_frontend/pages/exercises.eml]: does this belong here? looks unrelated to dark mode

File: src/ocamlorg_frontend/components/forms.eml [hunk 4]
```diff
@@ -25,7 +25,7 @@ let search_input ?button_attrs ?input_attrs ?dropdown_html ~label ~name _class =
 
 let select ~options ?attrs ?selected _class =
     <select
-        class="w-full lg:w-auto appearance-none text-legacy-default border rounded-md py-3 pl-6 border-gray-200 pr-14 h-12 <%s _class %>"
+        class="w-full lg:w-auto appearance-none text-title dark:text-dark-white border rounded-md py-3 pl-6 border-separator_20 dark:border-dark-separator_30 bg-white dark:bg-dark-card pr-14 h-12 <%s _class %>"
```
Comment (sabine) [inline src/ocamlorg_frontend/components/forms.eml]:
```suggestion
        class="w-full lg:w-auto appearance-none text-title dark:text-dark-white border rounded-lg py-3 pl-6 border-separator_20 dark:border-dark-separator_30 bg-white dark:bg-dark-card pr-14 h-12 <%s _class %>"
```

**Review by sabine — COMMENTED** [review]
Just a few remarks.

When the unrelated changes are removed, this is ready to merge.

File: asset/css/doc.css (hunk 1 above)
Comment (oyenuga17) [reply asset/css/doc.css]: I can't remember why I removed this, but I added it back now and everything is still good. Probably I was testing how some styles work and forgot to add it back

File: src/ocamlorg_frontend/pages/exercises.eml (hunk 3 above)
Comment (oyenuga17) [reply src/ocamlorg_frontend/pages/exercises.eml]: yes it's unrelated but I noticed there's no active state on the exercise menu so I added it here.

Comment (sabine) [reply src/ocamlorg_frontend/pages/exercises.eml]: good idea. Would be better if this was a separate PR because then this feature addition is self contained

Comment (oyenuga17) [reply src/ocamlorg_frontend/pages/exercises.eml]: yeah true. I'll create a separate pr for this then

**Comment by oyenuga17** [comment]
@sabine i think this  is good now

**Comment by sabine** [comment]
Almost, I wouldn't have expected the package search results page styles to change on light mode:

|old|this patch|
|-|-|
|Screenshot 2024-01-18 at 12-27-42 OCaml Packages · Search Result|Screenshot 2024-01-18 at 12-27-46 OCaml Packages · Search Result|

I think this is coming from the documentation search results design.

**Review by sabine — APPROVED** [review]
Merging!
